### ISSUE TITLE
S3 image upload doesn't work with ECS task IAM role

### ISSUE THREAD

**Issue by kshcherban** [grafana/grafana#12619]
In latest grafana (5.2.1) ECS task IAM role is not supported for image uploading. But EC2 role is.
You can check code here:
https://github.com/grafana/grafana/blob/v5.2.1/pkg/components/imguploader/s3uploader.go#L46-L54 

Not providing proper `access_key` and `secret_key` gives following error:
```
t=2018-07-16T12:57:23+0000 lvl=eror msg="Failed to upload alert panel image." logger=alerting.notifier error="NoCredentialProviders: no valid providers in chain. Deprecated.\n\tFor verbose messaging see aws.Config.CredentialsChainVerboseErrors"
```

Similar issue #9309 was fixed with CloudWatch access.

**Comment by torkelo** [comment]
@mtanda any ideas?

**Comment by mtanda** [comment]
Personally, I don't use ECS, but check it in this week.

**Comment by danielfariati** [comment]
I couldn't get it to work either, but for me the error is:
```
t=2018-08-09T22:14:31+0000 lvl=eror msg="Failed to upload alert panel image." logger=alerting.notifier error="AccessDenied: Access Denied\n\tstatus code: 403, request id: 021D9BAEEAD82A11, host id: zCv45kQa5x60ZfX/V4IKmOLynMPtNLhZIi1OJ9DMSRCS6uNc7WA648c3qJ7vFuentTYVVSfUp2Y="
```

I gave the ECS task full access to S3, to test it, but it continued not working.

**Comment by aidan-** [comment]
I believe this was fixed in #13424 (5.3, 5.3).

**Comment by kshcherban** [comment]
Yes, tested with 5.3.2, works now.
